### PR TITLE
view: reorganize planner components

### PR DESCRIPTION
**Pull request by TalDerei**
## Describe your changes
Reorganize the `ActionList` struct, enabling the web code to use the same structure when updating its rust dependencies. 
## Issue ticket number and link
References https://github.com/penumbra-zone/penumbra/issues/4332

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

### CODE REVIEW

**Comment by TalDerei** [comment]
@conorsch can you cherry pick this PR (on merge) into the [point release](https://github.com/penumbra-zone/penumbra/issues/4345?)?

**Review by hdevalence — APPROVED** [review]
This looks good but it won't be cherry-pickable onto an 0.73.x point release because it's built on top of breaking API changes. Instead we should include it in an 0.74.0 upgrade release.

File: crates/view/src/planner.rs [hunk 1]
```diff
@@ -455,6 +452,40 @@ impl<R: RngCore + CryptoRng> Planner<R> {
...
+    ///
+    /// Various logic is possible for note selection. Currently, this method
+    /// prioritizes notes sent to a one-time address, then notes with the largest
+    /// value:
+    ///
+    /// - Prioritizing notes sent to one-time addresses optimizes for a future in
+    /// which we implement DAGSync keyed by fuzzy message detection (which will not
+    /// be able to detect notes sent to one-time addresses). Spending these notes
+    /// immediately converts them into change notes, sent to the default address for
+    /// the users' account, which are detectable.
```
[the remark this answers is not in the document]
Comment (hdevalence) [reply crates/view/src/planner.rs:465]: I think this should still be the case? The method moved around but it’s still used right?

File: crates/core/transaction/src/action_list.rs [hunk 2]
```diff
@@ -230,46 +228,3 @@ impl ActionList {
...
-///
-/// Various logic is possible for note selection. Currently, this method
-/// prioritizes notes sent to a one-time address, then notes with the largest
-/// value:
-///
-/// - Prioritizing notes sent to one-time addresses optimizes for a future in
-/// which we implement DAGSync keyed by fuzzy message detection (which will not
-/// be able to detect notes sent to one-time addresses). Spending these notes
-/// immediately converts them into change notes, sent to the default address for
-/// the users' account, which are detectable.
```
Comment (cratelyn) [inline crates/core/transaction/src/action_list.rs:244]: i think i missed the discussion, but for posterity's sake: why are we no longer prioritize one-time addresses?

File: crates/view/src/planner.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cratelyn) [reply crates/view/src/planner.rs:465]: the method moved but the callback provided to `sort_by(..)` changed from

```rust
    filtered.sort_by(|a, b| {
        // Sort by whether the note was sent to an ephemeral address...
        match (
            a.address_index.is_ephemeral(),
            b.address_index.is_ephemeral(),
        ) {
            (true, false) => std::cmp::Ordering::Less,
            (false, true) => std::cmp::Ordering::Greater,
            // ... then by largest amount.
            _ => b.note.amount().cmp(&a.note.amount()),
        }
    });
```

to...

```rust
       filtered.sort_by(|a, b| b.note.amount().cmp(&a.note.amount()));
```

[the remark this answers is not in the document]
Comment (hdevalence) [reply crates/view/src/planner.rs:465]: Ah, got it, we should probably restore that logic then

[the remark this answers is not in the document]
Comment (TalDerei) [reply crates/view/src/planner.rs:465]: > Ah, got it, we should probably restore that logic then

Is this not the case anymore? https://discord.com/channels/824484045370818580/883455571003576320/1225512219677163601

File: crates/core/transaction/src/action_list.rs (hunk 2 above)
Comment (hdevalence) [reply crates/core/transaction/src/action_list.rs:244]: For instance, selecting the largest notes up-front doesn't necessarily reduce gas usage overall -- the smaller notes will remain unspent and will be spent later and the gas will be paid then. Spending all the small notes up front sweeps them into change, which might be desirable anyways.

File: crates/view/src/planner.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (TalDerei) [reply crates/view/src/planner.rs:465]: well the planners are independent implementations, so then maybe we just to filter for the largest notes in the web planner impl.

File: crates/core/transaction/src/action_list.rs (hunk 2 above)
Comment (TalDerei) [reply crates/core/transaction/src/action_list.rs:244]: that's true, good point

File: crates/view/src/planner.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cratelyn) [reply crates/view/src/planner.rs:465]: marking as resolved via 3294222

File: crates/core/transaction/src/action_list.rs (hunk 2 above)
Comment (cratelyn) [reply crates/core/transaction/src/action_list.rs:244]: marking as resolved via 3294222

**Pull request merged**